packages/core/src/TimeField/TimeFieldRoot.vue

```diff
@@ -0,0 +1,320 @@
+  disabled?: boolean
+  /** Whether or not the time field is readonly */
+  readonly?: boolean
+  /** Id of the element */
+  id?: string
+  /** The reading direction of the time field when applicable. <br> If omitted, inherits globally from `ConfigProvider` or assumes LTR (left-to-right) reading mode. */
+  dir?: Direction
+}
+
+export type TimeFieldRootEmits = {
+  /** Event handler called whenever the model value changes */
+  'update:modelValue': [date: TimeValue | undefined]
+  /** Event handler called whenever the placeholder value changes */
+  'update:placeholder': [date: TimeValue]
+}
+
+export const [injectTimeFieldRootContext, provideTimeFieldRootContext]
+  = createContext<TimeFieldRootContext>('TimeFieldRoot')
+
+function convertValue(value: TimeValue, date: DateValue = today(getLocalTimeZone())) {
+  if (value && 'day' in value) {
+    return value
+  }
+
+  return toCalendarDateTime(date, value)
+}
+</script>
+
+<script setup lang="ts">
+import { computed, nextTick, onMounted, ref, toRefs, watch } from 'vue'
+import { Primitive, usePrimitiveElement } from '@/Primitive'
+import { useVModel } from '@vueuse/core'
+import { VisuallyHidden } from '@/VisuallyHidden'
+
+defineOptions({
+  inheritAttrs: false,
+})
+
+const props = withDefaults(defineProps<TimeFieldRootProps>(), {
+  defaultValue: undefined,
+  disabled: false,
+  readonly: false,
+  placeholder: undefined,
+  isDateUnavailable: undefined,
+})
+const emits = defineEmits<TimeFieldRootEmits>()
+defineSlots<{
+  default: (props: {
+    /** The current time of the field */
+    modelValue: TimeValue | undefined
+    /** The time field segment contents */
+    segments: { part: SegmentPart, value: string }[]
+    /** Value if the input is invalid */
+    isInvalid: boolean
+  }) => any
+}>()
+
+const { disabled, readonly, granularity, defaultValue, minValue, maxValue, dir: propDir, locale: propLocale } = toRefs(props)
+const locale = useLocale(propLocale)
+const dir = useDirection(propDir)
+
+const formatter = useDateFormatter(locale.value)
+const { primitiveElement, currentElement: parentElement }
+  = usePrimitiveElement()
+const segmentElements = ref<Set<HTMLElement>>(new Set())
+
+const convertedMinValue = computed(() => minValue.value ? convertValue(minValue.value) : undefined)
+const convertedMaxValue = computed(() => maxValue.value ? convertValue(maxValue.value) : undefined)
+
+onMounted(() => {
+  getTimeFieldSegmentElements(parentElement.value).forEach(item => segmentElements.value.add(item as HTMLElement))
+})
+
+const modelValue = useVModel(props, 'modelValue', emits, {
+  defaultValue: defaultValue.value,
+  passive: (props.modelValue === undefined) as false,
+}) as Ref<TimeValue>
+
+const convertedModelValue = computed({
+  get() { return convertValue(modelValue.value) },
+  set(newValue) {
+    modelValue.value = modelValue.value && 'day' in modelValue.value ? newValue : new Time(newValue.hour, newValue.minute, newValue.second, modelValue.value?.millisecond)
```

Comment:
![image](https://github.com/user-attachments/assets/a34a3f02-8e3c-42e3-bb0c-064c8b35acdd)

throwing error when I clear the segment input